### ISSUE TITLE
Add a possibility to turn off weird html formatting

### ISSUE THREAD

**Issue by ilyakonrad** [prettier/prettier#12929]
The only thing that prevents us from using eslint+prettier in our Angular templates is https://github.com/prettier/prettier/issues/6460.

The rule `htmlWhitespaceSensitivity` **has** to have an option like "userSpecific" which would allow to have an adequate html formatting **and** a full control over whitespaces, while not having to bloat html with `<!-- prettier-ignore -->`.

It should be possible to have this html
![image](https://user-images.githubusercontent.com/25200186/170647897-e34ee994-821c-4301-bda6-ec1127b79682.png)
without having to turn it into this
![image](https://user-images.githubusercontent.com/25200186/170648069-84d73b34-0722-46fd-a6a7-99d315b85fb8.png)
or this
![image](https://user-images.githubusercontent.com/25200186/170648194-266d32bc-6151-48ec-b780-eac3df433230.png)

Here's another example of prettier complaining for no reason.
![image](https://user-images.githubusercontent.com/25200186/170648692-a6661dbe-a9eb-4bd4-826c-213a9100c4b3.png)
![image](https://user-images.githubusercontent.com/25200186/170649237-36b066be-b199-4638-9ef2-01598213d797.png)

I read all the points in favor of this ~ugliness~ design choice from the maintainers but I still hope it's possible to fix this and make prettier even better.

**Comment by Josh-Cena** [comment]
Can this be done heuristically? e.g. if there's a trailing punctuation, do not line break before?

**Comment by ilyakonrad** [comment]
> Can this be done heuristically? e.g. if there's a trailing punctuation, do not line break before?

You mean after? Wouldn't work, since it wouldn't fit.

![image](https://user-images.githubusercontent.com/25200186/170938860-5e42a0d5-118f-4b40-b0a1-94b0826dd93f.png)

**Comment by Josh-Cena** [comment]
No, I'm asking if _Prettier_ can make this change without introducing an option, not if there are workarounds under the status quo.

**Comment by ilyakonrad** [comment]
Any comments? @fisker

**Comment by fisker** [comment]
There are already several comments in #6460

**Comment by fisker** [comment]
If you are asking for comments on https://github.com/prettier/prettier/issues/12929#issuecomment-1140806965, as long as it won't break html, it should be acceptable.

**Comment by ilyakonrad** [comment]
Any progress a year later?
